haskell/yesod/README.md

```diff
@@ -0,0 +1,127 @@
+# yesod docker image
+
+Use this image as the base image for your yesod based applications.
+
+## Usage
+
+The simplest usage is to create a `Dockerfile` for your project that looks like
+this:
+```
+FROM thoughtbot/yesod:1.4.10-2
+
+RUN mkdir -p /app
+WORKDIR /app
+
+COPY *.cabal ./
+RUN cabal install --dependencies-only -j4 --enable-tests
+```
+
+## Setup Your Yesod Project
+
+First, create a `Dockerfile` for your project. The one shown above is a good
+starting place.
+
+Test that your container builds before moving on:
+```
+docker build .
+```
+
+If the build succeeds, it will output a alpha-numeric string as the final
+container name. Use this name in the next step.
+
+Now, try to use the resulting container like so:
+
+```
+docker run -iv $PWD:/app CONTAINER_NAME yesod devel
+```
+
+Yesod should compile the app and error out because of a missing postgresql
+database. We can setup postgres or other database dependencies use
+[docker-compose].
+
+[docker-compose]: https://docs.docker.com/compose/install/
+
+Create a `docker-compose.yml` file inside your project. A simple setup might
+look like this:
+
+```
+db:
+  image: postgres:9.4
+  ports:
+   - "5432"
+web:
+  build: .
+  command: yesod devel
+  environment:
+    - HOST=0.0.0.0
+    - PGUSER=postgres
+    - PGPASS
+    - PGHOST=db
+    - PGPORT=5432
+    - PGDATABASE=postgres
+  stdin_open: true
+  volumes:
+   - .:/app
+  ports:
+   - "3000:3000"
+  links:
+   - db
+```
+
+Build everything with `docker-compose build`.
+Run your project with `docker-compose up`.
+
+Your project should spin up and yesod will compile and run. You can now edit
+files on your machine and yesod will recompile every time the files change.
+
+### Testing
+
+You can run your tests anytime with `docker-compose run web yesod test`;
+however, this uses the development database to run the tests which may be
+undesirable. To add a second database for testing we can add these lines to our
+`docker-compose.yml` under the first `db` setup:
+
+```
+testdb:
+  image: postgres:9.4
+  ports:
+   - "5432"
+```
+
+Then add an new link to the `testdb` and a new ENV to our `web` process, making
+our final `docker-compose.yml` file look like:
+
+```
+db:
+  image: postgres:9.4
+  ports:
+   - "5432"
+testdb:
```

Comment:
I think creating the databases from the web container has replaced this testdb container, right?